markdown.dtx

```diff
@@ -24070,873 +24362,2812 @@ parsers.spnl                   = parsers.optionalspace
+%
+% Check if a trail exists and is non-empty in the indent table `indent_table`.
+%
+% \end{markdown}
+%  \begin{macrocode}
+local function has_trail(indent_table)
+  return indent_table ~= nil and 
+    indent_table.trail ~= nil and 
+    next(indent_table.trail) ~= nil
+end
 
+%  \end{macrocode}
+% \begin{markdown}
+%
+% Check if indent table `indent_table` has any indents.
+%
+% \end{markdown}
+%  \begin{macrocode}
+local function has_indents(indent_table)
+  return indent_table ~= nil and 
+    indent_table.indents ~= nil and 
+    next(indent_table.indents) ~= nil
+end
 
-parsers.raw_attribute = parsers.lbrace
-                      * parsers.optionalspace
-                      * parsers.equal
-                      * C(parsers.attribute_key)
-                      * parsers.optionalspace
-                      * parsers.rbrace
+%  \end{macrocode}
+% \begin{markdown}
+%
+% Add a trail `trail_info` to the indent table `indent_table`.
+%
+% \end{markdown}
+%  \begin{macrocode}
+local function add_trail(indent_table, trail_info)
+  indent_table.trail = trail_info
+  return indent_table
+end
 
--- block followed by 0 or more optionally
--- indented blocks with first line indented.
-parsers.indented_blocks = function(bl)
-  return Cs( bl
-         * (parsers.blankline^1 * parsers.indent * -parsers.blankline * bl)^0
-         * (parsers.blankline^1 + parsers.eof) )
+%  \end{macrocode}
+% \begin{markdown}
+%
+% Remove a trail `trail_info` from the indent table `indent_table`.
+%
+% \end{markdown}
+%  \begin{macrocode}
+local function remove_trail(indent_table)
+  indent_table.trail = nil
+  return indent_table
 end
-%    \end{macrocode}
-% \par
+
+%  \end{macrocode}
 % \begin{markdown}
 %
-%#### Parsers Used for Markdown Lists
+% Update the indent table `indent_table` by adding or removing a new
+% indent `add`.
 %
 % \end{markdown}
 %  \begin{macrocode}
-parsers.bulletchar = C(parsers.plus + parsers.asterisk + parsers.dash)
+local function update_indent_table(indent_table, new_indent, add)
+  indent_table = remove_trail(indent_table)
 
-parsers.bullet = ( parsers.bulletchar * #parsers.spacing
-                                      * (parsers.tab + parsers.space^-3)
-                 + parsers.space * parsers.bulletchar * #parsers.spacing
-                                 * (parsers.tab + parsers.space^-2)
-                 + parsers.space * parsers.space * parsers.bulletchar
-                                 * #parsers.spacing
-                                 * (parsers.tab + parsers.space^-1)
-                 + parsers.space * parsers.space * parsers.space
-                                 * parsers.bulletchar * #parsers.spacing
-                 )
+  if not has_indents(indent_table) then
+    indent_table.indents = {}
+  end
 
-local function tickbox(interior)
-  return parsers.optionalspace * parsers.lbracket
-       * interior * parsers.rbracket * parsers.spacechar^1
-end
+  local indent_table_indents = indent_table.indents
```

Comment:
Is there a purpose to this extra indirection? It certainly doesn't save us any characters. 😉